soc/espressif/esp32s3/memory.h

```diff
@@ -47,9 +48,22 @@
  */
 #define BOOTLOADER_USER_DRAM_END DRAM_BUFFERS_START
 
+/* Calculate required DRAM size for .bss.sector_buffers to adjust BOOTLOADER_DRAM_SEG_LEN
```

Comment:
I think it is much more straightforward to simply increase the `BOOTLOADER_DRAM_SEG_LEN` value to support the "worst case" scenario.
All the MCUboot segments - except the `BOOTLOADER_IRAM_LOADER` - will be rewritten by the last stage of the bootloader anyway.

Reply:
Simple is good 👍

In the worst-case scenario, we could have an ESP32S3 N32R8V with 32MB of flash. A crazy user might allocate approximately 16MB for the image partition. This would mean that sector_buffers might use around 0x10000 in the worst case.

So can i just add `0x10000` to `BOOTLOADER_DRAM_SEG_LEN` ?

Reply:
Since we can't support random "crazy user" let's be realistic now. The biggest partition size for the 32MB flash would be ~12MB, which is an addition of `0xC000` to the actual DRAM size, so with `0x15000` we should be fine.